### PR TITLE
Sound Update/Refactor

### PR DESCRIPTION
**Pull request by Fox-McCloud**
A number of sound updates from both TG and myself.

- Pre-defined channels are now  defines
- Start/stopping sounds on a specific channel is now a proc
- Fixed heartbeats stopping the ambience and buzz noise
- Heartbeats are now on their own channel
- Strips out environments from sound, due to their wonky nature
- Cleans up some sound code to make the code more compact and reduce redundancies
- Added ability to play sound that isn't impacted by pressure
- Added ability to specify a specific channel to play a sound on
- Altering a sound's frequency is done with a frequency var instead of pitch var
- Sounds are less likely to conflict with pre-defined channels (ambience, ship buzz, admin uploaded stuff, etc).
- Made a few loops that use `playsound_local` cheaper, by front-loading sound creation

### CODE REVIEW

File: code/modules/admin/verbs/playsound.dm [hunk 1]
```diff
@@ -6,7 +6,7 @@ var/list/sounds_cache = list()
 	if(!check_rights(R_SOUNDS))
 		return
 
-	var/sound/awful_sound = sound(null, repeat = 0, wait = 0, channel = 777)
+	var/sound/awful_sound = sound(null, repeat = 0, wait = 0, channel = CHANNEL_ADMIN)
```
Comment (Fox-McCloud) [inline code/modules/admin/verbs/playsound.dm]: the reason this doesn't use the new proc is so you're not creating a new null sound for everyone in the player list.

**Comment by KasparoVy** [comment]
Wicked good.

**Comment by Fox-McCloud** [comment]
After extensive testing, I've come to the conclusion that actual reverb is not possible with the current sound system without being hacky.

I'm immensely disappointed.

**Comment by Ataman** [comment]
> Altering a sound's frequency is done with a frequency var instead of pitch var
@Fox-McCloud Where did that come from? Everbody who works with sound immediately knows what a pitch variable will do. Unlike 'frequency' which is ambiguous. Are you setting the frequency irrespective of the original one? Are you shifting it now? How should we know?

**Comment by Fox-McCloud** [comment]
After extensive testing, I'm declaring echoes in SS13 dead until we get a better sound system.

`environment` applies globally to every sound, no matter how much you abuse the system, it will generate inconsistent results. Furthermore, any modification to `echo` doesn't yield any results unless you're already using an `environment` to begin with.

I can test out an environment of `1` or `2` with echoes, but needless to say, I doubt echoing will have much of an impact. 

I'm severely disappointed; I was really looking forward to echoey maintenance.

**Comment by Fox-McCloud** [comment]
I went ahead and made a few loops that used `playsound_local` cheaper.

File: code/game/objects/explosion.dm [hunk 2]
```diff
@@ -40,28 +40,22 @@
...
 				if(M && M.client)
 					var/turf/M_turf = get_turf(M)
 					if(M_turf && M_turf.z == epicenter.z)
 						var/dist = get_dist(M_turf, epicenter)
 						// If inside the blast radius + world.view - 2
 						if(dist <= round(max_range + world.view - 2, 1))
-							M.playsound_local(epicenter, get_sfx("explosion"), 100, 1, frequency, falloff = 5) // get_sfx() is so that everyone gets the same sound
+							M.playsound_local(epicenter, null, 100, 1, frequency, falloff = 5, S = explosion_sound)
 						// You hear a far explosion if you're outside the blast radius. Small bombs shouldn't be heard all over the station.
-						else if(dist <= far_dist)
```
Comment (Fox-McCloud) [inline code/game/objects/explosion.dm]: This looks like it changes a lot, but it doesn't; basically, if you were close to the explosion, it'd play the loud "boom"; if you were outside of the major blast radius, then you'd hear the muffled "boom".

That said, then, if you were slightly outside of this range, you'd hear a global soft "boom".

This seemed redundant; if everyone outside of the big blast radius is meant to hear a muffled boom, then why loop through the player list twice?

**Comment by Spacemanspark** [comment]
I'm not sure how I feel about echo maint being removed.

**Comment by Fox-McCloud** [comment]
At the setting it was using (`2`), you wouldn't hear any echo.

Secondly, it's not how the sound system actually works/should be used.

**Pull request merged**